spec/saxy/parser_spec.rb

```diff
@@ -4,16 +4,22 @@
   include FixturesHelper
 
   let(:parser) { Saxy::Parser.new(fixture_file("webstore.xml"), "product") }
+  let(:file_io) { File.new(fixture_file("webstore.xml")) }
 
-  it "should accept string filename as xml_file" do
+  it "should accept string filename for parsing" do
     xml_file = fixture_file("webstore.xml")
     parser = Saxy::Parser.new(xml_file, "product")
     parser.each.to_a.size.should == 2
   end
 
-  it "should accept IO as xml_file" do
-    xml_file = File.new(fixture_file("webstore.xml"))
-    parser = Saxy::Parser.new(xml_file, "product")
+  it "should accept IO for parsing" do
+    parser = Saxy::Parser.new(file_io, "product")
+    parser.each.to_a.size.should == 2
+  end
+
+  it "should accept an IO-like for parsing" do
+    allow(file_io).to receive(:is_a?).with(IO).and_return(false)
```

Comment:
Since the parser implementation has been changed and does not use `is_a?(IO)` internally anymore, this line has no effect and the test itself is actually a duplicate of the previous one (for string filename). I think we should rather try simulating an actual IO-like object here.

Not sure if that would be best testing practice, but maybe something like this:

```
class IOLike
  extend Forwardable
  def_delegators :@io, :read, :close

  def initialize(io)
    @io = io
  end
end

test "..." do
  parser = Saxy::Parser.new(IOLike.new(file_io), "product")
  ...
end
```